### ISSUE TITLE
`Run the 'configure' command first`

### ISSUE THREAD

**Issue by kremovtort** [kokobd/cabal-hoogle#4]
I have done everything from instruction after `cabal clean`. But when I run `cabal-hoogle generate` I'm getting following error message: `Run the 'configure' command first`.

**Comment by kokobd** [comment]
@Nexmean Have you run `cabal build` before `cabal-hoogle generate`? See: https://github.com/kokobd/cabal-hoogle#build-project-with-cabal

And, what Cabal version are you using? Can you share a link to your repository? Sharing the structure of your `dist-newstyle` may also help (for example with command like `tree`)

**Comment by kremovtort** [comment]
I did run `cabal build all` before `cabal-hoogle generate`. Sorry, but I can't share my repository. I can only share structure of `dist-newstyle` where I removed info about most modules and renamed some packages: https://gist.github.com/Nexmean/3e36772a320946dcbfbf1aeec5da93d2

**Comment by kokobd** [comment]
Please try not to disable the optimization, and see if it works.

From your `dist-newstyle` structure, I see docs reside in `noopt` subdirectory, indicating you were using something like `--disable-optimization` or `-O0` when calling `cabal build all`. However, `cabal-hoogle` currently only reads the default build product.

In the future, `cabal-hoogle` will call `cabal build` automatically, so that this will not happen.

**Comment by kremovtort** [comment]
Yeah, I have `optimization: False` in my `cabal.project.local`. So I'll try to disable it later and run it again.

**Comment by kremovtort** [comment]
I've tried to build w/o optimizations and now I get following error when call cabal-hoogle generate:
```console
$ cabal-hoogle generate
cabal-hoogle: Saved package config file is outdated:
• the Cabal version changed from Cabal-3.2.1.0 to Cabal-3.6.2.0
Re-run the 'configure' command.
```
cabal --version:
```console
$ cabal --version
cabal-install version 3.6.2.0
compiled using version 3.6.2.0 of the Cabal library
```

**Comment by kokobd** [comment]
This indicates some files are generated by Cabal-3.2.1.0.

1. Make sure you specified Cabal version when installing `cabal-hoogle`, for example: `cabal install --constraint="Cabal == 3.6.2.0" --install-method=copy --overwrite-policy=always cabal-hoogle`
2. Please try `rm -rf dist-newstyle`, and run `cabal build all`.

BTW, are you using a custom `Setup.hs`?

**Comment by kremovtort** [comment]
1. I'm sure that I specified Cabal version 3.6.2.0. I did it via `cabal.project.local` in local copy of cabal-hoogle repo:
```cabalproject
constraints:
   Cabal == 3.6.2.0
```
2. I did `cabal clean`. So ok, I can try `rm -rf dist-newstyle` instead.
3. Yeah, I'm using custom `Setup.hs` in one small package of project to get git revision.

**Comment by kokobd** [comment]
Ah, thank you for your patience! I see, packages with custom `Setup.hs` effectively use `Cabal` from the dependencies of `Setup.hs`. Maybe that version is 3.2 I guess.

I will:
1. Publish a version that allows partial failure in a multi-package repo very soon.
2. Look into how to use `Setup.hs` from `cabal-hoogle`

**Comment by kokobd** [comment]
@Nexmean I've merged a PR to allow partial failure in a multi-package project. You can try the latest `main` branch.

Meanwhile, a workaround to `Setup.hs` issue is by adding `Cabal == 3.6.2.0` constraint to **YOUR PROJECT**'s `cabal.project.local` too, so that your project's `Setup.hs` generates files for Cabal 3.6.2.0.

**Comment by kremovtort** [comment]
> I've merged a PR to allow partial failure in a multi-package project. You can try the latest main branch.

I've tried and it works for me.

> Meanwhile, a workaround to Setup.hs issue is by adding Cabal == 3.6.2.0 constraint to YOUR PROJECT's cabal.project.local too, so that your project's Setup.hs generates files for Cabal 3.6.2.0.

My build became broken when I set Cabal == 3.6.2.0, so I can't use it.

**Comment by kokobd** [comment]
Maybe worth noting that handling `Cabal` version mismatch when a custom `Setup.hs` is used is not easy. It's blocked by upstream. Created another issue to track that: #9
